db/db.go

```diff
@@ -66,7 +67,7 @@ func Open(path string, key tink.AEAD) (*DB, error) {
 
 // List returns secret metadata for all secrets on which at least one
 // member of 'from' has acl.ActionList permissions.
-func (db *DB) List(caps acl.Rules) ([]*api.SecretInfo, error) {
+func (db *DB) List(id audit.Principal, caps acl.Rules) ([]*api.SecretInfo, error) {
```

Comment:
I feel like we should put the audit principal in a standard place in the argument order. (If we think of it like context, maybe first? I don't have a strong opinion on where, but first or last seems fine)

Reply:
Moved both principal and permissions into an `ID` struct, and made it always be the first.

There's maybe an argument for server.go actually stuffing this information into a context.Context, and these DB methods checking the identity and permissions carried by the context... Maybe I'll play with that in a future PR.

Reply:
I thought about suggesting that, but since we don't (yet) have contexts here, I didn't want to add a bunch of additional plumbing yet. I agree that once we do have contexts, though, that would be a good home for this type.